lib/tasks/attachments/delete_attachment.rake

```diff
@@ -0,0 +1,33 @@
+desc "Delete and unpublish attachment, redirect to parent URL"
+task :delete_attachment, %i[attachment_content_id] => :environment do |_, args|
+  attachments = Attachment.where(content_id: args[:attachment_content_id], deleted: false)
+  if attachments.empty?
+    puts "Unable to find any non-deleted attachments with content_id #{args[:attachment_content_id]}"
+    next
+  end
+
+  attachments.each do |attachment|
+    edition = attachment.attachable
+
+    unless edition&.state == "superseded"
+      puts "Edition does not exist or edition is not superseded."
+      puts "If edition is not superseded, please get user to delete attachment through UI."
+      puts "This rake task is only used in situations where we are unable to remove via UI"
+      next
+    end
+
+    puts "Deleting attachment: #{attachment.title}..."
+    attachment_data = attachment.attachment_data
+    attachment.destroy!
+    ServiceListeners::AttachmentUpdater.call(attachment_data:)
+
+    puts "Unpublishing attachment: #{attachment.title}..."
+    edition.translations.each do |translation|
+      PublishingApiRedirectWorker.new.perform(
+        attachment.content_id,
+        edition.public_path,
+        translation.locale.to_s,
+      )
+    end
```

Comment:
I note that the destroy command from the `AttachmentsController` doesn't hunt down all the translations and redirect the attachment for each of them - is that why we're not removing attachments from public access successfully? If so, should we fix that, and also push this down the stack so that it can be called both by the controller and this class?

Reply:
I did think about this - but it requires a bit more thought than fixing this bug right now. 
At the moment, the attachments get a "deleted" flag set when it's deleted via the controller. and only at publish/republish time when it goes through the worker, does it handle attachments and sync up publishing API. I feel like changing this process now is going to take a bit more time, and probably a more chunky trello card.

Reply:
Right, that does make sense, we don't want to immediately remove the attachment from GOV.UK. Trying to maintain aggregate documents is hard. Thanks for explaining